Directory.Build.props

```diff
@@ -60,4 +69,12 @@
     <!-- Optional: Build symbol packages (.symbols.nupkg) to distribute the PDB containing Source Link -->
     <IncludeSymbols>true</IncludeSymbols>
   </PropertyGroup>
+
+  <!-- Work around https://github.com/NuGet/Home/issues/6091 
+       TODO: Determine if this is still needed-->
+  <ItemDefinitionGroup>
+    <PackageReference>
+      <PrivateAssets>None</PrivateAssets>
+    </PackageReference>
+  </ItemDefinitionGroup>
```

Comment:
Add a tracking issue for the TODO?

Reply:
Thank you for this reminder.  I meant to address it here.  Let me test it and see if there is any difference.

Reply:
This does in fact create a nuspec diff.  If I remove this all PackageReferences change like the following:
```diff
-<dependency id="LightGBM" version="3.3.5" include="All" />
+<dependency id="LightGBM" version="3.3.5" exclude="Build,Analyzers" />
```

So this would cause any transitive dependency's `Build` assets to no longer apply.  `Analyzers` are not impacted because nuget isn't responsible for those.

However - it wasn't impacting ProjectReferences - so the only time this would really make a difference was when an external packageReference was brought in and the customer referenced the ML.NET package which referenced that package - if there was an indirect reference through another ML.NET package it wouldn't help since that reference would still have exclude=build.

I went through all unique package references to examine if their content to see if they had build assets that would apply.
You can see the analysis here: https://gist.github.com/ericstj/0108a09e702a274a17108e49dc2e81cd?permalink_comment_id=4946135#gistcomment-4946135

The only two packages with `build` assets and no `buildTransitive` were `LightGBM` and `Microsoft.ML.OnnxRuntime.Managed`.  We can opt in these package references individually, and also follow up with those package authors to use the `buildTransitive` folder for their targets.

I've applied the inclusion to just those PackageReferences.

Filed issues to encourage those package owners to use buildTransitive.
https://github.com/microsoft/LightGBM/issues/6344
https://github.com/microsoft/onnxruntime/issues/19676